components/base/sensorcontrolled/movestraight.go

```diff
@@ -34,22 +34,36 @@ func (sb *sensorBase) MoveStraight(
 	// If a position movement sensor or controls are not configured, we cannot use this MoveStraight method.
 	// Instead we need to use the MoveStraight method of the base that the sensorcontrolled base wraps.
 	// If there is no valid velocity sensor, there won't be a controlLoopConfig.
-	if sb.position == nil || len(sb.controlLoopConfig.Blocks) == 0 {
+	if len(sb.controlLoopConfig.Blocks) == 0 {
 		sb.logger.CWarnf(ctx,
-			"Position reporting sensor not available, or control loop not configured, using base %s's MoveStraight",
+			"control loop not configured, using base %s's MoveStraight",
 			sb.controlledBase.Name().ShortName())
 		if sb.loop != nil {
 			sb.loop.Pause()
 		}
 		return sb.controlledBase.MoveStraight(ctx, distanceMm, mmPerSec, extra)
 	}
+	if sb.position == nil {
+		sb.logger.CWarn(ctx,
+			"controlling using linear velocity only, for increased accuracy add a position reporting sensor")
+
+		// adjust inputs to ensure errDist is always positive to match the position based implementation
+		if distanceMm < 0 {
+			distanceMm = -distanceMm
+			mmPerSec = -mmPerSec
+		}
```

Comment:
hmm, if I input both a negative distance and a negative speed, doesn't that change the direction of the speed as defined by the wheeled base driver?

we're expecting positive net energy in that case. We can also just pretend physics don't exist.

Reply:
But I was writing the API contract and the expectation of two negative inputs is going in the +ve directionality for our robotics system.

Reply:
I was treating a negative distance & speed as a request to drive forward. this check + some other logic in this code handles that request